### ISSUE TITLE
Unexpected/inconsistent port forward on vcluster creation

### ISSUE THREAD

**Issue by jandress** [loft-sh/vcluster#650]
### What happened?

Some interesting vcluster weirdness today. At first, I thought this was something in one of my configs that I had sideways, but this is two snippets from a scripted vcluster creation one right after the other with no changes made to anything.

The first run:
```
+ kubectl create namespace my-vcluster
namespace/my-vcluster created
+ vcluster create my-vcluster -n my-vcluster -f values.yaml
info   Detected local kubernetes cluster minikube. Will deploy vcluster with a NodePort & sync real nodes
info   Create vcluster my-vcluster...
info   execute command: helm upgrade my-vcluster https://charts.loft.sh/charts/vcluster-0.11.0.tgz --kubeconfig /tmp/3824438060 --namespace my-vcluster --install --repository-config='' --values /tmp/635985927 --values values.yaml
done √ Successfully created virtual cluster my-vcluster in namespace my-vcluster
info   Waiting for vcluster to come up...
done √ Switched active kube context to vcluster_my-vcluster_my-vcluster_minikube
warn   Since you are using port-forwarding to connect, you will need to leave this terminal open
- Use CTRL+C to return to your previous kube context
- Use `kubectl get namespaces` in another terminal to access the vcluster
Forwarding from 127.0.0.1:11966 -> 8443
Forwarding from [::1]:11966 -> 8443
^Cinfo   Switched back to context minikube

```
And the second run:
```
+ kubectl create namespace my-vcluster
namespace/my-vcluster created
+ vcluster create my-vcluster -n my-vcluster -f values.yaml
info   Detected local kubernetes cluster minikube. Will deploy vcluster with a NodePort & sync real nodes
info   Create vcluster my-vcluster...
info   execute command: helm upgrade my-vcluster https://charts.loft.sh/charts/vcluster-0.11.0.tgz --kubeconfig /tmp/1350817504 --namespace my-vcluster --install --repository-config='' --values /tmp/1682711150 --values values.yaml
done √ Successfully created virtual cluster my-vcluster in namespace my-vcluster
info   Waiting for vcluster to come up...
done √ Switched active kube context to vcluster_my-vcluster_my-vcluster_minikube
- Use `vcluster disconnect` to return to your previous kube context
- Use `kubectl get namespaces` to access the vcluster
+ kubectl apply -f nginxdemo.yaml
```

### What did you expect to happen?

I'm using nodeport for this, so I do not expect to be getting the port forward at all. 

### How can we reproduce it (as minimally and precisely as possible)?

I am running the following in a shell script on minikube on an EC2 instance:

kubectl create namespace my-vcluster
vcluster create my-vcluster -n my-vcluster -f values.yaml
kubectl apply -f nginxdemo.yaml

### Anything else we need to know?

Discussed this with @FabianKramm on slack:

[Fabian Kramm](https://app.slack.com/team/UDUJVB4U8)
  [8 minutes ago](https://loft-sh.slack.com/archives/C01N273CF4P/p1659739729574119?thread_ts=1659737524.595299&cid=C01N273CF4P)
Mhh seems that the vcluster wasn't reachable in time after creation there, which leads vcluster to fallback to port-forwarding

[Fabian Kramm](https://app.slack.com/team/UDUJVB4U8)
  [8 minutes ago](https://loft-sh.slack.com/archives/C01N273CF4P/p1659739747206959?thread_ts=1659737524.595299&cid=C01N273CF4P)
We could add a flag to force vcluster to wait if that would help you

### Host cluster Kubernetes version

<details>
```
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.3", GitCommit:"aef86a93758dc3cb2c658dd9657ab4ad4afc21cb", GitTreeState:"clean", BuildDate:"2022-07-13T14:30:46Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.1", GitCommit:"3ddd0f45aa91e2f30c70734b175631bec5b5825a", GitTreeState:"clean", BuildDate:"2022-05-24T12:18:48Z", GoVersion:"go1.18.2", Compiler:"gc", Platform:"linux/amd64"}
```
</details>

### Host cluster Kubernetes distribution

<details>

```
minikube version: v1.26.0
commit: f4b412861bb746be73053c9f6d2895f12cf78565

```

</details>

### vlcuster version

<details>
```
vcluster version 0.11.0
```
</details>

### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
k3s
```

</details>

### OS and Arch

<details>

```
OS: 
cat /etc/*release*
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04 LTS"
PRETTY_NAME="Ubuntu 22.04 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy

Arch:
uname -a
Linux ip-10-0-0-10 5.15.0-1015-aws #19-Ubuntu SMP Wed Jun 22 17:44:56 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

```
</details>

**Comment by FabianKramm** [comment]
@jandress thanks for creating this issue! Seems like waiting times out which is why vcluster falls back to port-forwarding, we need a better way to determine if the vcluster is reachable there

**Comment by jandress** [comment]
@FabianKramm in my particular case, falling back to the port forwarding hangs my script, so it is preferrable to fail out entirely rather than use port forwarding. Ideally, I'd like to see a switch to tell it to wait as long as it takes. Additionally, a warning like "something is broken, falling back to port forwarding" when not using the switch would be really handy. When I was first sorting the automation out for this, this issue was completely indistinguishable from having misconfigured networking somewhere and I spent a lot of time chasing myself in small circles over it.

**Comment by FabianKramm** [comment]
@jandress we released v0.11.1 yesterday which should fix this problem without a flag, would you mind testing again?

**Comment by jandress** [comment]
@FabianKramm I've done a bit of testing on the new release and have not seen the issue pop up again yet, even when running a set of actions which would previously surface the issue every time. I am cautiously optimistic :)
